### PR TITLE
[feature][doc] Add docs for system topic and update for topic policy

### PR DESCRIPTION
**Pull request by momo-jun**
Add docs for #15619 and also fix issue #13856.

### Modifications

1. Add docs for `system topic`.
2. Improve `topic-level policies` section: add notes for required configs and add an illustration.

The preview looks good, and pls note that there is a slight difference between the latest (master) and earlier versions.

**Latest**
Since 2.11, `system topic` defaults to `enabled` and support listings with a new flag.
image
image
image

**2.10 and earlier versions**
1. Add a note for manual configs to enable system topic.
image
2. The note has slight difference with that of master.
image

### Documentation
  
- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @RobertIndie @codelipenghui @MarvinCai @Anonymitaet can you pls review this doc PR? Thanks.

**Review by Anonymitaet — COMMENTED** [review]
Is it better to reverse the arrow?
image

Why broker1 does not have steps 1,2,5?

**Comment by momo-jun** [comment]
> Is it better to reverse the arrow? image
> 
> Why broker1 does not have steps 1,2,5?

1. This is an emphasis that brokers subscribe to the system topic proactively to receive change events.
2. Only broker0 gets the request from pulsar-admin to implement the change, but the change applies to all brokers in a single namespace.

**Review by codelipenghui — COMMENTED** [review]
I think we'd better add a new page/section for pulsar system topics, system topics are just a topic name internally in Pulsar, `__change_events` is one of them.

1. What is the system topic in Pulsar
2. Why are we using system topic
3. How many system topics that Pulsar have
4. How to enable system topic
5. System topic limitations

**Comment by momo-jun** [comment]
> I think we'd better add a new page/section for pulsar system topics, system topics are just a topic name internally in Pulsar, `__change_events` is one of them.
> 
> 1. What is the system topic in Pulsar
> 2. Why are we using system topic
> 3. How many system topics that Pulsar have
> 4. How to enable system topic
> 5. System topic limitations

@codelipenghui Makes sense. Who can provide the draft?

**Comment by momo-jun** [comment]
> > I think we'd better add a new page/section for pulsar system topics, system topics are just a topic name internally in Pulsar, `__change_events` is one of them.
> > 
> > 1. What is the system topic in Pulsar
> > 2. Why are we using system topic
> > 3. How many system topics that Pulsar have
> > 4. How to enable system topic
> > 5. System topic limitations
> 
> @codelipenghui Makes sense. Who can provide the draft?

@Technoboy- will provide more inputs soon.

**Comment by momo-jun** [comment]
Ping @Technoboy- @mattisonchao for review.

**Review by Technoboy- — APPROVED** [review]

**Comment by momo-jun** [comment]
Ping @Anonymitaet  for review.

**Review by Anonymitaet — APPROVED** [review]

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -857,6 +857,38 @@ Producer<byte[]> producer = client.newProducer()
...
+System topic is a predefined topic for internal use within Pulsar. It can be either persistent or non-persistent topic.
+
+System topics serve to implement certain features and eliminate dependencies on third-party components, such as transactions, heartbeat detections, topic-level policies, and resource group services. System topics empower the implementation of these features to be simplified, dependent, and flexible. Take heartbeat detections for example, you can leverage the system topic for healthcheck to internally enable producer/reader to procude/consume messages under the heartbeat namespace, which can detect whether the current service is still alive.
+
+There are diverse system topics depending on namespaces. The following table outlines the available system topics for each specific namespace.
+
+| Namespace | TopicName | Domain | Count | Usage |
+|-----------|-----------|--------|-------|-------|
+| pulsar/system | `transaction_coordinator_assign_${id}` | Persistent | Default 16 | Transaction coordinator |
+| pulsar/system | `_transaction_log${tc_id}` | Persistent | Default 16 | Transaction log |
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply site2/docs/concepts-messaging.md:871]: yes, right.

[the remark this answers is not in the document]
Comment (momo-jun) [reply site2/docs/concepts-messaging.md:871]: I will update it to `_transaction_log_${tc_id}` in a follow-up PR. Thank you both.

[the remark this answers is not in the document]
Comment (michaeljmarshall) [reply site2/docs/concepts-messaging.md:871]: Thank you for updating it and for adding the documentation!

**Pull request merged**